### PR TITLE
cleanup some logic of nodelifecycle contoller

### PR DESCRIPTION
**Pull request by lucming**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by lucming** [comment]
@alculquicondor @smarterclayton please help me to review the changes,thanks

File: pkg/util/taints/taints.go [hunk 1]
```diff
@@ -288,15 +288,13 @@ func TaintKeyExists(taints []v1.Taint, taintKeyToMatch string) bool {
 func TaintSetDiff(t1, t2 []v1.Taint) (taintsToAdd []*v1.Taint, taintsToRemove []*v1.Taint) {
 	for _, taint := range t1 {
 		if !TaintExists(t2, &taint) {
-			t := taint
-			taintsToAdd = append(taintsToAdd, &t)
+			taintsToAdd = append(taintsToAdd, &taint)
```
Comment (alculquicondor) [inline pkg/util/taints/taints.go:291]: This was intentional.

Here for reference:
https://medium.com/swlh/use-pointer-of-for-range-loop-variable-in-go-3d3481f7ffc9

File: pkg/util/taints/taints.go [hunk 2]
```diff
@@ -288,15 +288,13 @@ func TaintKeyExists(taints []v1.Taint, taintKeyToMatch string) bool {
...
 		}
 	}
 
 	for _, taint := range t2 {
 		if !TaintExists(t1, &taint) {
-			t := taint
-			taintsToRemove = append(taintsToRemove, &t)
+			taintsToRemove = append(taintsToRemove, &taint)
```
Comment (alculquicondor) [inline pkg/util/taints/taints.go:297]: this too.

File: pkg/controller/nodelifecycle/node_lifecycle_controller.go [hunk 3]
```diff
@@ -1385,22 +1385,24 @@ func (nc *Controller) addPodEvictorForNewZone(node *v1.Node) {
...
-		} else {
-			nc.zoneNoExecuteTainter[zone] =
-				scheduler.NewRateLimitedTimedQueue(
-					flowcontrol.NewTokenBucketRateLimiter(nc.evictionLimiterQPS, scheduler.EvictionRateLimiterBurst))
-		}
-		// Init the metric for the new zone.
-		klog.Infof("Initializing eviction metric for zone: %v", zone)
-		evictionsNumber.WithLabelValues(zone).Add(0)
-		evictionsTotal.WithLabelValues(zone).Add(0)
+	if _, found := nc.zoneStates[zone]; found {
```
Comment (alculquicondor) [inline pkg/controller/nodelifecycle/node_lifecycle_controller.go:1388]: This code is supposed to go away in the next release (1.26) #111411
I would just leave it

**Review by alculquicondor — COMMENTED** [review]
/priority backlog

I would leave this for release 1.26

File: pkg/controller/nodelifecycle/node_lifecycle_controller.go (hunk 3 above)
Comment (lucming) [reply pkg/controller/nodelifecycle/node_lifecycle_controller.go:1388]: > the

okay, i'll close the pr then, thanks for your reply.

Comment (lucming) [reply pkg/controller/nodelifecycle/node_lifecycle_controller.go:1388]: > This code is supposed to go away in the next release (1.26) #111411 I would just leave it

okay, i'll close the pr then, thanks for your reply.

**Comment by alculquicondor** [comment]
The changes in `pkg/controller/util/node/controller_utils.go` are useful IMO

**Pull request closed without merging**